### ISSUE TITLE
[AWS Batch on EKS] [request]: For Batch on EKS, how to use Kubernetes secrets for Batch EKS Jobs?

### ISSUE THREAD

**Issue by dushyant8858** [aws/containers-roadmap#2045]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Tell us about your request**
For Batch on EKS, how to use Kubernetes secrets for Batch EKS Jobs?

**Which service(s) is this request for?**
AWS Batch EKS

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
For Batch on EKS, how to use Kubernetes secrets for Batch EKS Jobs?
```
spec:
  containers:
    - envFrom:
        - secretRef:
            name: 
```
Currently, EKS Batch only supports these parameter https://docs.aws.amazon.com/batch/latest/APIReference/API_EksContainer.html
i.e I see an option to specify secret volume into the container https://docs.aws.amazon.com/batch/latest/APIReference/API_EksSecret.html
but does not see an option to specify Kubernetes secret as an Env into the Batch EKS Jobs. 

**Are you currently working around this issue?**
I could not find any work around to specify Kubernetes secret as an Env into the Batch EKS Jobs

**Comment by nboisnea** [comment]
I have a similar issue with `valueFrom` inside of `env`. In a k8s YAML pod configuration I can simply inject a secret into an environment variable. The only way to inject secrets in an EKS job definition is with volumeMounts, which is inconvenient. EksPodProperties is too restrictive. It should accept all standard k8s options and apply them to the pod as is.

**Comment by se-meiser** [comment]
Yes, would be nice if this would not be restricted to secrets,  I would like to set an env variable to something like 
```
valueFrom:
  fieldRef:
    fieldPath: status.hostIP
```

**Comment by Paulius0112** [comment]
Is anyone working on this issue?
